syntax = "proto3";

package vendor;

service VendorService {
    rpc GetApiConfig(ApiConfigRequest) returns (APIConfiguration);
    rpc GetCommandTemplate(CommandTemplateRequest) returns (CommandTemplate);
}

import "google/protobuf/struct.proto";

message CommandTemplateRequest {
    string api_config_id = 1;
    string command_type = 2;
}

message CommandTemplate {
    string url_template = 1;
    string method = 2;
    google.protobuf.Struct headers_template = 3;
    google.protobuf.Struct body_template = 4;
    repeated string required_params = 5;
    repeated string optional_params = 6;
}

message ApiConfigRequest {
    string api_config_id = 1;
}

message APIConfiguration {
    string id = 1;
    string vendor_id = 2;
    string name = 3;
    string version = 4;
    string description = 5;
    string base_url = 6;

    // Authentication Configuration
    string auth_type = 7;
    google.protobuf.Struct auth_config = 8;

    // Request Templates
    google.protobuf.Struct headers_template = 9;

    // Settings
    int32 timeout = 10;
    int32 retry_count = 11;
    int32 retry_delay = 12;
}